### PR TITLE
Contains DrugGroup class, Service, ServiceImpl, DAO, HibernateDAO, updated liquidbase, etc.

### PR DESCRIPTION
**Pull request by diegodayan**
Added: DrugGroup.java
Added: DrugGroupService.java
Added: DrugGroupServiceImpl.java
Added: DrugGroupDAO.java
Added: HibernateDrugGroupDAO.java
Added: RaxacoreDrugGroup.hbm.xml
Added: DrugGroupController.java
Added: DrugGroupResource.java
Updated: liquidbase.xml
Updated: moduleApplicationContext.xml
Updated: config.xml

### CODE REVIEW

**Comment by johnstoecker** [comment]
Hi Diego -- at a first pass this looks good. Can you add some JUnit tests on the DAO just for sanity? Also, were you able to build + output .omod file in the target folder + upload to OpenMRS as a module?

File: omod/src/main/java/org/raxa/module/raxacore/web/v1_0/controller/DrugGroupController.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+import org.raxa.module.raxacore.DrugGroup;
+import org.raxa.module.raxacore.DrugGroupService;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+
+@Controller
+@RequestMapping(value = "/rest/v1/raxacore/druggroup")
+public class DrugGroupController extends BaseRestController {
+	
+	DrugGroupService service = Context.getService(DrugGroupService.class);
```
Comment (johnstoecker) [inline omod/src/main/java/org/raxa/module/raxacore/web/v1_0/controller/DrugGroupController.java:27]: There will be some issues if we initialize our service here --  see this commit on how to initialize the service: https://github.com/Raxa/raxacore/commit/dd23ca987fa99631c5e5e1dfad5e1e8c6b1958c0

File: omod/src/main/java/org/raxa/module/raxacore/web/v1_0/resource/DrugGroupResource.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+	public List<DrugGroup> getAllDrugGroupList() {
+		return getDrugGroupService().getDrugGroupList();
+	}
+	
+	@Override
+	public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+		if (rep instanceof DefaultRepresentation) {
+			DelegatingResourceDescription description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			//		description.addProperty("display", findMethod("getDisplayString"));
```
Comment (johnstoecker) [inline omod/src/main/java/org/raxa/module/raxacore/web/v1_0/resource/DrugGroupResource.java:42]: Remove all the extra commented lines.

File: api/src/main/java/org/raxa/module/raxacore/DrugGroup.java [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+	
+	private Integer id;
+	
+	private String drugGroupName;
+	
+	private Date dateCreated;
+	
+	private String uuid;
+	
+	private String searchQuery;
```
Comment (johnstoecker) [inline api/src/main/java/org/raxa/module/raxacore/DrugGroup.java:20]: We don't need to save a user's searchQuery -- we can delete all references to searchquery here and below.

File: api/src/main/java/org/raxa/module/raxacore/DrugInfo.java [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+ */
+public class DrugInfo extends BaseOpenmrsMetadata implements Serializable {
+	
+	private Integer drugInfoId;
+	
+	private Integer drugId;
+	
+	private double price;
+	
+	private String searchQuery;
```
Comment (johnstoecker) [inline api/src/main/java/org/raxa/module/raxacore/DrugInfo.java:34]: We don't need to save the search query for drug_info, either.

File: api/src/main/java/org/raxa/module/raxacore/DrugInfoService.java [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+/*
+ * Interface for interacting with the DrugInfo
+ */
+@Transactional
+public interface DrugInfoService extends OpenmrsService {
+	
+	public void setDrugInfoDAO(DrugInfoDAO dao);
+	
+	// @Authorized({"Add Patient Lists"}) // TODO: Permissions
```
Comment (johnstoecker) [inline api/src/main/java/org/raxa/module/raxacore/DrugInfoService.java:36]: For all your permissions, you will want to add "Add Drug Info" "Edit Drug Info" "Delete Drug Info" "View Drug Info" or similar permissions for drug group. These permissions will go into the config.xml

File: api/src/main/java/org/raxa/module/raxacore/DrugGroupService.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+import java.util.List;
+import org.openmrs.api.OpenmrsService;
+import org.openmrs.annotation.Authorized;
+import org.openmrs.util.PrivilegeConstants;
+import org.raxa.module.raxacore.db.DrugGroupDAO;
+import org.springframework.transaction.annotation.Transactional;
+
+public interface DrugGroupService extends OpenmrsService {
+
```
Comment (johnstoecker) [inline api/src/main/java/org/raxa/module/raxacore/DrugGroupService.java:11]: Once we have our linking tables complete, we want to get the drugs from inside our drug_group:
1. Add new methods in our service here that return List<Drug>
2. Add the drugs full resource into what is returned in the controller

Comment (johnstoecker) [inline api/src/main/java/org/raxa/module/raxacore/DrugGroupService.java:11]: *number 1 should be: return List<Drug>

File: api/src/main/java/org/raxa/module/raxacore/impl/DrugGroupServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
+package org.raxa.module.raxacore.impl;
+
+import java.util.List;
+import org.openmrs.api.impl.BaseOpenmrsService;
+import org.raxa.module.raxacore.DrugGroup;
+import org.raxa.module.raxacore.DrugGroupService;
+import org.raxa.module.raxacore.db.DrugGroupDAO;
+import org.springframework.transaction.annotation.Transactional;
+
+public class DrugGroupServiceImpl extends BaseOpenmrsService implements DrugGroupService {
```
Comment (johnstoecker) [inline api/src/main/java/org/raxa/module/raxacore/impl/DrugGroupServiceImpl.java:10]: I don't believe you need 'extends BaseOpenmrsService' here

File: api/src/main/java/org/raxa/module/raxacore/impl/DrugInfoServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+import org.openmrs.api.context.Context;
+import org.raxa.module.raxacore.DrugInfo;
+import org.raxa.module.raxacore.DrugInfoService;
+import org.raxa.module.raxacore.db.DrugInfoDAO;
+
+/*
+ * Implements PatientListService.java Note the PatientList query must be in the
+ * form of:
+ * "?encounterType=<uuid>&startDate=2012-05-07&endDate=2012-05-08&inlist=<uuidForList>&notinlist=<uuidForList>"
+ */
```
Comment (johnstoecker) [inline api/src/main/java/org/raxa/module/raxacore/impl/DrugInfoServiceImpl.java:40]: Can delete all PatientList comments

Comment (johnstoecker) [inline api/src/main/java/org/raxa/module/raxacore/impl/DrugInfoServiceImpl.java:40]: Do a search on 'PatientList' and change all relevant comments, so our javadocs will reference the right object

**Comment by johnstoecker** [comment]
Just some changes here and there before the pull request looks good. It might be best to finish DrugGroup fully before adding the DrugInfo files to this pull request. 

I would also like to know whether you are able to actually make REST calls on the druggroup resource when added to your local openmrs instance.

**Comment by diegodayan** [comment]
Both PUT/POST for /raxacore/druggroup are working. We are currently having issues with th JUnit tests. I am closing this pull request and will open a new one when things look better.

**Pull request closed without merging**